core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -123,6 +123,25 @@ public fun CoroutineScope.peripheral(
     )
 }
 
+public fun CoroutineScope.peripheral(
+    macAddress: String,
+    builderAction: PeripheralBuilderAction = {},
+): Peripheral? {
+    val builder = PeripheralBuilder()
+    builder.builderAction()
+    val bluetoothDevice = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(macAddress)
+    if (bluetoothDevice.type == BluetoothDevice.DEVICE_TYPE_UNKNOWN) return null // DEVICE_TYPE_UNKNOWN indicates that the device may have been cleared from the cache
+    return AndroidPeripheral(
+        coroutineContext,
+        bluetoothDevice,
+        builder.transport,
+        builder.phy,
+        builder.observationExceptionHandler,
+        builder.onServicesDiscovered,
+        builder.logging,
+    )
+}
```

Comment:
This function can piggyback on the function with `BluetoothDevice` in its signature.

```suggestion
public fun CoroutineScope.peripheral(
    identifier: Identifier,
    builderAction: PeripheralBuilderAction = {},
): Peripheral {
    val bluetoothDevice = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(identifier)
    return peripheral(bluetoothDevice, builderAction)
}
```